src/frequenz/dispatch/_bg_service.py

```diff
@@ -219,6 +220,12 @@ async def new_running_state_event_receiver(
 
     # pylint: enable=redefined-builtin
 
+    @override
+    async def stop(self, msg: str | None = None) -> None:
+        """Stop the background service."""
+        self._next_event_timer.stop()
```

Comment:
We are probably going to deprecate `.stop()` as it was something specific for `Timer`, but now we have `close()` in the `Receiver` interface, so it might be more future-proof to use `close()`.


Also since this is a background service, it is better to put the `close()` in `cancel()` instead of `stop()`, otherwise if you `d.cancel(); await d` the timer won't be closed.

Finally this might be more complicated than you thought, at least we had issues in some other places where now you start getting a `ReceiverStoppedError`/`StopAsyncIteration` exception in the `run` you might need to start catching. Maybe this is a more actor-specific issue because of the auto-restart and is not really an issue for pure background services though, but just something to have in mind.